test/integration/integration.ts

```diff
@@ -142,63 +210,72 @@ function makeTrustLine(testcase, address, secret) {
   };
   const trust = client
     .prepareTrustline(address, specification, {})
-    .then((data) => {
+    .then(async (data) => {
       const signed = client.sign(data.txJSON, secret);
-      if (address === wallet.getAddress()) {
+      if (address === walletAddress) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call -- Cleaner to read without additional typing
         testcase.transactions.push(signed.id);
       }
       return client.request({
         command: "submit",
         tx_blob: signed.signedTransaction,
       });
     })
-    .then(() => ledgerAccept(client));
+    .then(async () => ledgerAccept(client));
   return trust;
 }
 
-function makeOrder(client, address, specification, secret) {
+// eslint-disable-next-line max-params -- It's worth the extra parameter to simplify the logic
+async function makeOrder(
+  client: Client,
+  address: string,
+  specification: FormattedOrderSpecification,
+  secret: string
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- The return type is legitimately any
+): Promise<any> {
   return client
     .prepareOrder(address, specification)
     .then((data) => client.sign(data.txJSON, secret))
-    .then((signed) =>
+    .then(async (signed) =>
       client.request({ command: "submit", tx_blob: signed.signedTransaction })
     )
-    .then(() => ledgerAccept(client));
+    .then(async () => ledgerAccept(client));
 }
 
-function setupAccounts(testcase) {
-  const client = testcase.client;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- The return type is legitimately any
+async function setupAccounts(testcase: Mocha.Context): Promise<any> {
```

Comment:
Can you rework the logic of this like you did with some of the other methods, so it's easier to read?